rest-new/src/main/java/org/jboss/pnc/rest/endpoints/AbstractEndpoint.java

```diff
@@ -0,0 +1,92 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.pnc.rest.endpoints;
+
+import org.jboss.pnc.dto.DTOEntity;
+
+import org.jboss.pnc.dto.response.Page;
+import org.jboss.pnc.facade.providers.api.Provider;
+import org.jboss.pnc.facade.validation.ConflictedEntryException;
+import org.jboss.pnc.rest.configuration.SwaggerConstants;
+import org.jboss.pnc.facade.validation.DTOValidationException;
+import org.jboss.pnc.rest.api.parameters.PageParameters;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+import javax.validation.constraints.NotNull;
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.NotFoundException;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Response;
+
+
+public class AbstractEndpoint<DTO extends REF, REF extends DTOEntity> {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    private Provider<?, DTO, REF> provider;
+
+    private Class<DTO> dtoClass;
+
+    @Deprecated
+    public AbstractEndpoint() {
+    }
+
+    public AbstractEndpoint(Provider<?, DTO, REF> provider, Class<DTO> dtoClass) {
+        this.provider = provider;
+        this.dtoClass = dtoClass;
+    }
+
+    protected DTO create(DTO dto) {
+        logger.debug("Creating an entity with body: " + dto);
+        dto = provider.store(dto);
+        logger.debug("Entity with id: " + dto.getId() + " successfully created");
+        return dto;
+    }
+
+    protected DTO getSpecific(int id) {
+        logger.debug("Getting " + dtoClass.getSimpleName() + " with id: " + id);
+        DTO dto = provider.getSpecific(id);
+        if (dto == null) {
+            logger.debug("Entity of type " + dtoClass.getSimpleName() + " with id: " + id + " not found.");
+            throw new NotFoundException("Entity of type " + dtoClass.getSimpleName() + " with id: " + id + " not found.");
+        }
+        logger.debug("Successful retrieval of " + dtoClass.getSimpleName() + " with id: " + id);
+        return dto;
+    }
+
+    protected Page<DTO> getAll(@NotNull PageParameters pageParameters) {
+        logger.debug("Retrieving " + dtoClass.getSimpleName() + "s with these " + pageParameters);
+        return provider.getAll(pageParameters.getPageIndex(), pageParameters.getPageSize(), pageParameters.getSort(), pageParameters.getQ());
+    };
+
+    protected void update(int id, DTO dto) {
+        logger.debug("Updating " + dtoClass.getSimpleName() + " with id: " + id);
+        provider.update(id, dto);
+        return;
+    }
+
+    protected void delete(int id) {
+        logger.debug("Deleting " + dtoClass.getSimpleName() + " with id: " + id);
+        provider.delete(id);
+        logger.debug("Deletion of " + dtoClass.getSimpleName() + " with id: " + id + " was successful");
+        return;
```

Comment:
Return statements at the end of void class is pointless.

Reply:
Yeah, there was a comment before that told what happens when after that return but wasn't needed after. Thx for the catch.